packages/block-editor/src/components/inserter-list-item/index.js

```diff
@@ -83,7 +83,7 @@ function InserterListItem( {
 						disabled={ item.isDisabled }
 						onClick={ ( event ) => {
 							event.preventDefault();
-							onSelect( item );
+							onSelect( item, event.metaKey );
```

Comment:
I implemented the "ctrl + click" behavior but it only works for mouse clicks for now (and not "Enter" clicks), I wonder if it has something to do with `CompositeItem` component cc @diegohaz 

As far as I know the "Button" block is not doing anything special with the onKeyDown prop, so I'm assuming `CompositeItem` is the one responsible for actually calling `onClick` as a response to `Enter` key down? (I'm not entirely sure though)

Reply:
I fixed this by adding an explicit onKeyDown handler.

Reply:
Yeah! `CompositeItem` will call `element.click()` on <kbd>Enter</kbd> and <kbd>Space</kbd> when the element is not a native button. Since it's just `.click()`, no state from the keydown/keyup event is passed over to the click event. We could create a mouse event instead and pass all the modifier key state together with it. I'll create an issue there proposing this.

But, yeah, you should be able to achieve the same result by adding an explicit `onKeyDown`.